### PR TITLE
dvtm: refactor (fix on darwin)

### PR DESCRIPTION
**Pull request by luke-clifton**
###### Motivation for this change

This PR adds the required CFLAGS to get dvtm to build in darwin. It also applies a patch which has been sitting idle upstream for a while. This patch is required for dvtm to work on newer versions of macOS.

See https://github.com/martanne/dvtm/pull/69

The patch works fine on Linux as well.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by luke-clifton** [comment]
@vrthra are you able to review this?

File: pkgs/tools/misc/dvtm/default.nix [hunk 1]
```diff
@@ -28,7 +39,6 @@ stdenv.mkDerivation rec {
     description = "Dynamic virtual terminal manager";
     homepage = http://www.brain-dump.org/projects/dvtm;
     license = licenses.mit;
-    platforms = platforms.linux;
```
Comment (LnL7) [inline pkgs/tools/misc/dvtm/default.nix]: Don't remove platforms, use `platforms.unix` or `platforms.linux ++ platforms.darwin`

**Comment by joachifm** [comment]
@GrahamcOfBorg build dtvm

**Comment by joachifm** [comment]
@GrahamcOfBorg build dvtm

**Comment by joachifm** [comment]
Thank you

**Pull request merged**